src/app.rs

```diff
@@ -101,17 +101,14 @@ impl <'a> UI <'a> {
 
     pub fn result_handler(&mut self, result: String) {
         let number = result.trim().parse::<f64>();
-        let current_time = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .unwrap()
-            .as_secs_f64();
+        let elapsed_seconds = (Instant::now() - self.start_time).as_secs_f64();
```

Comment:
I think it would be better to completly avoid storing duration using `f64` but I tried to be as less intrusive at possible :)
Let me know if you agree with a systematic usage of Duration.

Reply:
I agree the more we stay with Duration the better.
Though, I think it might be better to continue storing the timestamps as floats in the results dataset to avoid a conversion each time we compute the regression.

Don't worry about being intrusive and thanks for the help! :)

Reply:
> Though, I think it might be better to continue storing the timestamps as floats in the results dataset to avoid a conversion each time we compute the regression.


I would say this might be a premature optimisation ;)
I think it is too bad to avoid `Duration`, and it may be surprising but I think `Duration` store it's value as an integer, so it may be better since it may reduce the count of arithmetic operations with floating point, but it's pure speculation.

BTW I just found out Godbold support rust!! :)
<img width="1440" alt="Screen Shot 2019-12-27 at 00 40 07" src="https://user-images.githubusercontent.com/685196/71503046-9241ea00-2841-11ea-9534-136f046b2f4a.png">

Reply:
I agree we should try to use Duration and see how the perf is impacted. If not working or too slow I think I'll do a custom linear regression.

Do you want to do it in this PR ?

Reply:
I will try to do it.